internal/smdclient/oidc.go

```diff
@@ -0,0 +1,36 @@
+package smdclient
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+)
+
+// Structure of a token reponse from OIDC server
+type oidcTokenData struct {
+	Access_token string `json:"access_token"`
+	Expires_in   int    `json:"expires_in"`
+	Scope        string `json:"scope"`
+	Token_type   string `json:"token_type"`
+}
+
+// Refresh the cached access token, using the provided JWT server
+func (s *SMDClient) RefreshToken() error {
```

Comment:
One thing to keep in mind is that OPAAL's token endpoint returns a JWT bearer token without having to perform an authorization grant that's expected from OAuth 2 servers. I may be worth implementing some way to do a client credentials grant (which is what OPAAL is doing here) to get a token later on. For now though, this should be fine.